File: prime_test.rs
mod prime;

#[test]
fn test_prime_sieve_get_primes() {
    let mut is_prime_array = [true; 20];

    prime::prime_sieve_initialize(&mut is_prime_array);
    let actual = prime::prime_sieve_get_primes(&is_prime_array).collect::<Vec<_>>();

    let expected = vec![2, 3, 5, 7, 11, 13, 17, 19];
    assert_eq!(actual, expected);
}

File: prime_test.py
from prime import get_is_prime_array, get_primes_from_is_prime_array

def test_get_primes_from_is_prime_array():
    is_prime_array = get_is_prime_array(10)
    actual = list(get_primes_from_is_prime_array(is_prime_array))
    assert(actual == [2, 3, 5, 7])